cfy_manager/components/postgresql_server/postgresql_server.py

```diff
@@ -1732,15 +1739,90 @@ def start(self):
             service.verify_alive(POSTGRES_SERVICE_NAME)
         logger.notice('PostgreSQL Server successfully started')
 
+    def _save_encoding_and_locale(self):
+        self.encoding = subprocess.run(
+            ['psql', '-U', 'postgres', '-c', 'SHOW SERVER_ENCODING', '-t'],
+            stdout=subprocess.PIPE, stderr=subprocess.DEVNULL
+        ).stdout.decode('utf-8').strip()
+
+        self.locale = subprocess.run(
+            ['psql', '-U', 'postgres', '-c', 'SHOW LC_CTYPE', '-t'],
+            stdout=subprocess.PIPE, stderr=subprocess.DEVNULL
+        ).stdout.decode('utf-8').strip()
+
     def stop(self, force=True):
+        self._save_encoding_and_locale()
         logger.notice('Stopping PostgreSQL Server...')
         if config[POSTGRESQL_SERVER]['cluster']['nodes']:
             service.stop('etcd')
             service.stop('patroni')
         else:
-            service.stop(POSTGRES_SERVICE_NAME)
+            try:
+                service.stop(POSTGRES_SERVICE_NAME)
+                self.upgrading_psql = False
+            except ProcessExecutionError:
+                service.stop(OLD_POSTGRES_SERVICE_NAME)
+                service.remove(OLD_POSTGRES_SERVICE_NAME)
+                self.upgrading_psql = True
         logger.notice('PostgreSQL Server successfully stopped')
 
+    def upgrade(self):
+        if self.upgrading_psql:
```

Comment:
maybe `if not` and early return? one level of indentation less for the whole method